google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -313,4 +324,26 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
 
     assertNull(ServiceOptions.getServiceAccountProjectId(credentialsFile.getPath()));
   }
+
+  @Test
+  public void testResponseHeaderContainsMetaDataFlavor() throws Exception {  
```

Comment:
Great test! I think it might be simpler to mock the `HttpRequest` itself (e.g. `when(request).execute().thenReturn(mockResponse)`.